### PR TITLE
Feature/18 implement profile UI

### PR DESCRIPTION
**Pull request by kzepfl**
## Changes

- Implemented UI for `ProfileScreen` (`Profile.kt`)

_The score, profile picture and QRcode are only placeholder_

- Added `ProfileScreen` to Navigation (`MainActivity.kt`)

- Implemented a squeleton of Bottom Navigation menu (`BottomNavigationMenu.kt`)

_Malick_ has already made one so mine is only a place holder

## Testing

- Implemented UI tests for the `ProfileScreen` composable. (`ProfileScreenTest.kt`)

_The_ test verify that all element are displayed, will test the navigation in the future.

### CODE REVIEW

**Comment by kzepfl** [comment]
**Wait for Feature/14 implement map UI to be merged before this one** (I have my own bottom navigation bar, i will merge on my branche beforehand so their will be no conflict)

File: app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            tabList = LIST_TOP_LEVEL_DESTINATION,
+            selectedItem = navigationActions.currentRoute())
+      },
+      content = { padding ->
+        Column(
+            modifier = Modifier.fillMaxSize().padding(padding).testTag("ProfileColumn"),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.spacedBy(10.dp, Alignment.CenterVertically)) {
+              // profile placeholder
+              Row() {
```
Comment (misterM125) [inline app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt:45]: Suggestion: I fail to see the utility of using rows for everyone of your nested composables. This does not add anything to the UI and increases code complexity for no reason, I would suggest removing them.
(this concerns line 45, 52, 59, 66 and 74)

File: app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+              // QR code placeholder
+              Row() {
+                Image(
+                    painter = painterResource(id = R.drawable.qrcode),
+                    contentDescription = "qr code",
+                    modifier = Modifier.size(260.dp))
+              }
+              // button to add a new friend
+              Row() {
+                Button(
```
Comment (misterM125) [inline app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt:67]: Improvement: The two buttons (lines 67 and 75) feel a bit too close to each other, making the UI feel a bit too cluttered, I suggest adding a spacer in between the qr code and the first button, as well as in between both buttons:

`Spacer(modifier = Modifier.height(10.dp))`

This should add a bit more clarity to your ui and make the buttons a bit more easy to click on

File: app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+              // button to train with a friend
+              Row() {
+                Button(
+                    onClick = {},
+                    colors = ButtonDefaults.buttonColors(Color(0xFFA53A36)),
+                    modifier = Modifier.size(220.dp, 50.dp).testTag("profileTrainButton")) {
+                      Text(text = "Train with a friend", fontSize = 17.sp)
+                    }
+              }
+              Spacer(modifier = Modifier.height(100.dp))
```
Comment (misterM125) [inline app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt:82]: Suggestion: this last spacer does not feel useful especially when the Composables are already centered inside the scaffold, I would suggest removing it

**Review by misterM125 — CHANGES_REQUESTED** [review]
Overall, the code is very good, tests are extensive and the UI is very pleasant to look,
I would just suggest a few changes to improve this UI even more and make your code more clear

File: app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt (hunk 1 above)
Comment (kzepfl) [reply app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt:45]: Indeed, I removed the row

File: app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt (hunk 3 above)
Comment (kzepfl) [reply app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt:82]: It was to make the UI more similar to the Figma as it was a bit higher

File: app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt (hunk 2 above)
Comment (kzepfl) [reply app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt:67]: Done

**Review by misterM125 — APPROVED** [review]
Perfect! Ui looks great, thanks for applying the changes

**Pull request merged**